pdns/auth-caches.cc

```diff
@@ -45,7 +47,31 @@ uint64_t purgeAuthCaches(const std::string& match)
   return ret;
 }
 
-/* remove specific entries from all caches, no wildcard matching */
+/* remove zone from all caches */
+uint64_t purgeAuthCacheForZone(const DNSName& name)
+{
+  uint64_t ret = 0;
+
+  if (g_zoneCache.isEnabled()) {
+    DomainInfo di;
+    UeberBackend B;
+    if (B.getDomainInfo(name, di, false)) {
```

Comment:
This is not free, and almost always unnecessary. For example, the 2136 code that calls this even already has the di for you.